src/Tablebot/Plugins/Quote.hs

```diff
@@ -64,14 +76,41 @@ showQ qId m = do
       sendMessageVoid m $ pack $ txt ++ " - " ++ author
     Nothing -> sendMessageVoid m "Couldn't get that quote!"
 
+-- | @deleteQuote@, which looks for a message of the form @!quote delete n@,
+-- and removes it from the database.
+deleteQ :: Int64 -> Message -> DatabaseDiscord ()
+deleteQ qId m =
+  requirePermission Any m $
```

Comment:
Possible idea: change the type to `requirePermission :: RequiredPermission -> (Message -> DatabaseDiscord ()) -> Message -> DatabaseDiscord ()`, thus allowing you to define `deleteQ` as follows:
```
deleteQ qID = requirePermission Any $ \m -> let k....
```
where `requirePermission` takes the message, checks it and then passes it to the continuation if the permission holds.

Alternatively, once errors are merged in, if `requirePermission` throws then you won't need a continuation at all, since throwing breaks control flow.